cmd/zdb/zdb.c

```diff
@@ -1568,6 +1568,15 @@ snprintf_blkptr_compact(char *blkbuf, size_t buflen, const blkptr_t *bp,
 		if (bp_freed)
 			(void) snprintf(blkbuf + strlen(blkbuf),
 			    buflen - strlen(blkbuf), " %s", "FREE");
+		if (dump_opt['d'] > 6) {
```

Comment:
Looking at this again I think it's unfortunate that we need to include (another) a `dump_opt` check in `snprintf_blkptr_compact`.  I don't think it would be unreasonable for us to add the checksum to the 6 d's output.  That of course does change the output format, but since `zdb` is a developer facing tool I think we have some more latitude there.  It's _possible_ this would break some existing ZTS tests.  Thoughts?

Reply:
Didn't know tests don't like dump_opt...
One idea would be to always use such verbosity or just have a simple boolean flag: All verbose or not... No multiple levels, take all verbosity or none

Reply:
To clarify, using `dump_opt` in a helper function like this which formats output is what I found a little troubling.  It's not a huge problem, but something I felt worth mentioning and considering in the context of this PR.

Reply:
I'd also prefer including the checksum at 6d 's and dropping dump_opt here. I wouldn't want to add another d for every new output format change. At some point (arguably already arrived at) it just gets ridiculous. Breaking existing tests and scripts was my concern for that approach, but those could just be fixed when discovered.

Reply:
just going 6d's and droping dump_opt would be the most clean solution... Yes some issues might be caused, but issues are there to be fixed :)